platform-sdk/swirlds-base/src/main/java/com/swirlds/base/ArgumentUtils.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.base;
+
+import java.util.Objects;
+
+/**
+ * Class that contains common checks like null checks as static methods.
+ */
+public class ArgumentUtils {
+
+    /**
+     * Private constructor to prevent instantiation.
+     */
+    private ArgumentUtils() {}
+
+    /**
+     * Throw an {@link NullPointerException} if the supplied argument is {@code null}.
+     *
+     * @param argument     the argument to check
+     * @param argumentName the name of the argument
+     */
+    public static <T> T throwArgNull(final T argument, final String argumentName) {
```

Comment:
Add `@NotNull` to return type and both parameters.

Reply:
argumentName can be null, but you are right: annotations are missing

Reply:
how can the name be null - see line 39.  More important: why should it ever be null?  can't the programmer be bothered to supply it?  doesn't he have an emacs macro or something?

Reply:
Annotations will be added once https://github.com/hashgraph/hedera-services/pull/5481 is merged.

Reply:
@david-bakin-sl if it is `null` "null" will be part of the message. it makes no sense but technically null for the name will throw no exception.